private-media-framework/draft-ietf-perc-private-media-framework.md

```diff
@@ -711,3 +698,362 @@ a co-author.
+utility.  To reduce complexity, PERC prescribes that endpoints
+manufacturer random SRTP master keys locally to be used for E2E
+encryption.
+
+Since the media server is not involved in E2E functions, it will not
+create this key.  Note, too, that even the key distributor is unaware
+of the locally-generated E2E keys used by each endpoint.
+
+## Who has What Key
+
+To summarize the various keys and which entity is in possession
+of a given key, refer to (@fig-who-has-what-key).
+
+All endpoints have knowledge of the KEK and it is the same for
+all participants.
+
+Every HBH key is distinct at a given endpoint, thus Endpoint A and
+endpoint B do not have knowledge of the other's HBH key.
+
+Each endpoint generates its own E2E Key (SRTP master key), thus the
+key distinct per endpoint.  This key is transmitted (encrypted) via
+the EKT Field to other endpoints.  Endpoints that receive media from
+a given transmitting endpoint will have knowledge of the transmitter's
+E2E key.
+
+{#fig-who-has-what-key align="center"}
+~~~
++----------------------+------------+-------+-------+------------+
+| Key     /    Entity  | Endpoint A |  MD X |  MD Y | Endpoint B |
++----------------------+------------+-------+-------+------------+
+| KEK                  |    Yes     |  No   |  No   |     Yes    |
++----------------------+------------+-------+-------+------------+
+| E2E Key (A and B)    |    Yes     |  No   |  No   |     Yes    |
++----------------------+------------+-------+-------+------------+
+| HBH Key (A<=>MD X)   |    Yes     |  Yes  |  No   |     No     |
++----------------------+------------+-------+-------+------------+
+| HBH Key (B<=>MD Y)   |    No      |  No   |  Yes  |     Yes    |
++----------------------+------------+---------------+------------+
+| HBH Key (MD X<=>MD Y)|    No      |  Yes  |  Yes  |     No     |
++----------------------+------------+---------------+------------+
+~~~
+Figure: Keys per Entity
+
+# PERC Packet Format {#packetformat}
+
+This appendix is intended to aid the reader in understanding the PERC media
+packet format by walking through the steps required to construct the packet
+at each step.
+
+## PERC Packet Formation Steps
+
+### Step 1: Formulate an RTP Packet
+
+First, we start with a plaintext RTP packet that is formed by the
+transmitting endpoint, but has not undergone any kind of cryptographic
+transform.  It would look like the one shown in [@fig-plaintext-packet].
+
+{#fig-plaintext-packet align="center"}
+~~~
++------------------------------+
+|          RTP Header          |
++------------------------------+
+|                              |
+|       Plaintext Payload      |
+|                              |
++------------------------------+
+~~~
+Figure: Plaintext RTP Packet before Encryption
+
+While not explicitly called out, the "RTP Header" in the diagrams in this
+appendix would include any allowable or required RTP header extensions.
+
+### Step 2: Apply the E2E Cryptographic Transform
+
```

Comment:
I think this way of describing it in two steps is not right for the framework. Imagine we defined a form of double that took the part of the packet that was protected with the only half of the key and put it in the even bits and took the part that needed to be done with the full key and put them in the odd bits. This would be a lousy way to design a double transform but it work fine with the framework and is a perfectly legal way to write a SRTP transform. The problem with describing it like it is hear is that it is not really constant with the SRTP abstraction of what a transform is. So while it is somewhat true for the current version of double, it's not true in general and it forces one to know all the details of how every part work instead of the framework using a SRTP as a modular block that has extensible transforms and any transform with some certain properties would work fine for use in this framework.